src/core/server/plugins/plugins_service.ts

```diff
@@ -107,6 +121,27 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
     await this.pluginsSystem.stopPlugins();
   }
 
+  private generateUiPluginsConfigs(
+    uiPlugins: Map<string, DiscoveredPlugin>
+  ): Map<PluginName, Observable<unknown>> {
+    return new Map(
+      [...uiPlugins].map(([pluginId, plugin]) => {
+        const configDescriptor = this.pluginConfigDescriptors.get(pluginId);
+        if (configDescriptor && configDescriptor.exposeToBrowser) {
+          return [
+            pluginId,
+            this.configService
+              .atPath(plugin.configPath)
+              .pipe(
+                map((config: any) => pick(config || {}, configDescriptor.exposeToBrowser || []))
+              ),
+          ];
+        }
+        return [pluginId, of({})];
```

Comment:
is this necessary with your check here? https://github.com/elastic/kibana/pull/50641/commits/daf09e3be50a80716baf2698d8a2275b4f8c792b#diff-ce5eebb41b6315c38446e8a68a4ca0aeR242

Reply:
I was filtering null values after the map call using `[...uiPlugins].map([...]).filter`, but it seems typescript doesn't properly recognize  `Observable<unknown> | null` filtered with `!== null` as `Observable<unknown>`. So the only purpose of this is to avoid having the method return type to `Map<PluginName, Observable<unknown> | null>`. But this can be changed